rust_src/src/marker.rs

```diff
@@ -603,6 +611,26 @@ impl LispBufferRef {
 
 // Converting between character positions and byte positions.
 
+// When converting bytes from/to chars, we look through the list of
+// markers to try and find a good starting point (since markers keep
+// track of both bytepos and charpos at the same time).
+// But if there are many markers, it can take too much time to find a "good"
+// marker from which to start.  Worse yet: if it takes a long time and we end
+// up finding a nearby markers, we won't add a new marker to cache this
+// result, so next time around we'll have to go through this same long list
+// to (re)find this best marker.  So the further down the list of
+// markers we go, the less demanding we are w.r.t what is a good marker.
+// The previous code used INITIAL=50 and INCREMENT=0 and this lead to
+// really poor performance when there are many markers.
+// I haven't tried to tweak INITIAL, but experiments on my trusty Thinkpad
+// T61 using various artificial test cases seem to suggest that INCREMENT=50
+// might be "the best compromise": it significantly improved the
+// worst case and it was rarely slower and never by much.
+// The asymptotic behavior is still poor, tho, so in largish buffers with many
+// overlays (e.g. 300KB and 30K overlays), it can still be a bottleneck.
+static BYTECHAR_DISTANCE_INITIAL: isize = 50;
+static BYTECHAR_DISTANCE_INCREMENT: isize = 50;
```

Comment:
Where do we need this ?

Reply:
I guess it's intended to be used somewhere in `LispBufferRef::charpos_to_bytepos`, but isn't.